File: Services/ObjectManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace SeaBan
{
    class ObjectManager
    {
        private static List<SceneObject> sceneObjects = new List<SceneObject>();

        public static void clear()
        {
            sceneObjects.Clear();
        }

        public static SceneObject get(int index)
        {
            if (index > sceneObjects.Count) return null;
            else
                return sceneObjects[index];
        }

        public static int getIndexByObject(SceneObject sObject)
        {

            for (int i = 0; i < sceneObjects.Count; i++)
            {
                if (sceneObjects[i] == sObject) return i;
            }

            return -1;
        }


        public static void del(int index)
        {
            try
            {
                if (index > sceneObjects.Count) return;
                else
                    sceneObjects.RemoveAt(index);
            }
            catch { }
        }


        public static int getCount()
        {
            return sceneObjects.Count;
        }

        public static void add(SceneObject sceneObject)
        {
            sceneObjects.Add(sceneObject);
        }


    }
}

File: Old_JavaVersion/SeaBan/src/com/dci/seaban/Objects/ObjectManager.java
package com.dci.seaban.Objects;

import java.util.ArrayList;
import java.util.List;

import com.dci.seaban.Render.SceneRender;

import android.content.Context;
import android.opengl.GLES20;

public final class ObjectManager {
	private static final List<SceneObject> sceneObject = new ArrayList<SceneObject>();
	
	public static final void clear()
	{
		sceneObject.clear();
	}
	
	public static final SceneObject get(int index)
	{		
		if (index > sceneObject.size()) return null;
		else 
			return sceneObject.get(index); 
	}
	
	public static final int getIndexByObject(SceneObject sObject)
	{		
		
			for (int i=0; i < sceneObject.size(); i++)
			{
		 	  if (sceneObject.get(i) == sObject) return i;
			}
			
			return -1;
	}
	
	
	public static final void del(int index)
	{		
		if (index > sceneObject.size()) return;
		else 
			sceneObject.remove(index); 
	}
	
	
	public static final int getCount()
	{
		return sceneObject.size();
	}
	
	public static final void add(SceneObject object){
		sceneObject.add(object);
	}

	
}
